From b88f1096ad88c8ee3a88b6d483387c4e6eab6f61 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Tue, 9 Jun 2020 20:06:50 +0530
Subject: [PATCH] meson: Export symbols for a DLL when not building statically

`default_library` might be 'both', in which case we will fail to link
when building with MSVC.
---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index 1238e0b6e..483741047 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -383,7 +383,7 @@ version = '0.' + '0'.join(meson.project_version().split('.')) + '.0'
 
 extra_hb_cpp_args = []
 if cpp.get_id() == 'msvc'
-  if get_option('default_library') == 'shared'
+  if get_option('default_library') != 'static'
     extra_hb_cpp_args += '-DHB_DLL_EXPORT'
   endif
   hb_so_version = ''
-- 
2.27.0

